native/core/src/execution/planner.rs

```diff
@@ -992,14 +994,25 @@ impl PhysicalPlanner {
                 let predicate =
                     self.create_expr(filter.predicate.as_ref().unwrap(), child.schema())?;
 
-                let filter = Arc::new(FilterExec::try_new(
-                    predicate,
-                    Arc::clone(&child.native_plan),
-                )?);
-                Ok((
-                    scans,
-                    Arc::new(SparkPlan::new(spark_plan.plan_id, filter, vec![child])),
-                ))
+                if filter.use_datafusion_filter {
```

Comment:
If someone can show me how to do a ternary or match here, that would be awesome. I tried to do `let filter = if use_datafusion_filter`... but because I was possibly assigning one of two different types (`DataFusionFilterExec` or `CometFilterExec`) the compiler wasn't happy when I went to pass that into the `SparkPlan::new` call.